esp/src/src-react/components/Filter.tsx

```diff
@@ -116,12 +116,18 @@ const TargetClusterTextField: React.FunctionComponent<TextFieldProps> = (props)
         });
     }, []);
 
-    return <TextField {...props} >
-        {targetClusters.map(tc => <MenuItem key={tc.Name} value={tc.Name}>{tc.Name}{tc.Name !== tc.Type ? ` (${tc.Type})` : ""}</MenuItem>)}
-    </TextField>;
+    return <Dropdown
+        {...props}
+        options={targetClusters.map(n => {
```

Comment:
This map should happen within the useEffect above (I know I had done the same previously) - but it can help with performance.